[WPE] Fix compilation issue with USE_GBM=OFF and USE_LIBDRM=OFF

zhani opened the pull request:
#### 0ec85924c7860453cef53c541ee5d743707cc2e7
<pre>
[WPE] Fix compilation issue with USE_GBM=OFF and USE_LIBDRM=OFF
<a href="https://bugs.webkit.org/show_bug.cgi?id=285229">https://bugs.webkit.org/show_bug.cgi?id=285229</a>

Reviewed by Carlos Garcia Campos.

When USE_GBM=OFF and USE_LIBDRM=OFF, WPE platform compilation fails.
This combination also implies ENABLE_GPU_PROCESS=OFF. The patch
adds necessary ifdefs and includes to resolve the issue.

* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:

Canonical link: <a href="https://commits.webkit.org/288832@main">https://commits.webkit.org/288832@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eeb8698c4910f4329544df885192ebfe377ea24e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/84460 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/4085 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38765 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/89542 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/35471 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/4170 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/12069 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/65682 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23526 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/87506 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/3146 "Found 3 new test failures: fast/files/blob-stream-frame.html http/tests/websocket/tests/hybi/no-subprotocol.html http/wpt/cache-storage/quota-third-party.https.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76736 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45976 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/3097 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/34519 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/73980 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31739 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90922 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11727 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8545 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/74135 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11954 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72560 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/73337 "Found 1 new API test failure: /WebKitGTK/TestWebViewEditor:/webkit/WebKitWebView/cut-copy-paste/non-editable (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17675 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16122 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/3158 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13168 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11679 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/17155 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/11528 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/15004 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/13301 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
```diff
@@ -181,7 +181,7 @@ static const char* openGLAPI()
 }
 
-#if PLATFORM(GTK) || (PLATFORM(WPE) && ENABLE(WPE_PLATFORM))
+#if PLATFORM(GTK) || (PLATFORM(WPE) && USE(LIBDRM) && ENABLE(WPE_PLATFORM))
```
I know it's existing code, but we don't really need PLATFORM(WPE) && ENABLE(WPE_PLATFORM) since WPE_PLATFORM is only defined for WPE. Also, LIBDRM is optional for gtk too, so we should first check LIBDRM and then platforms

zhani replied:
There was already another commit in upstream fixing this issue so this change is obsolete

carlosgcampos commented on Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
```diff
@@ -534,7 +534,7 @@ void WebKitProtocolHandler::handleGPU(WebKitURISchemeRequest* request)
             addTableRow(hardwareAccelerationObject, "Buffer format"_s, renderBufferFormat(request));
         }
-#elif PLATFORM(WPE) && ENABLE(WPE_PLATFORM)
+#elif PLATFORM(WPE) && USE(LIBDRM) && ENABLE(WPE_PLATFORM)
```
Same here

zhani replied:
There was already another commit in upstream fixing this issue so this change is obsolete

carlosgcampos commented on Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
```diff
@@ -168,7 +168,7 @@ Vector<DMABufRendererBufferFormat> WebPageProxy::preferredBufferFormats() const
 }
 
-#if ENABLE(WPE_PLATFORM)
+#if PLATFORM(WPE) && USE(GBM) && ENABLE(WPE_PLATFORM)
```
So, no need to add the PLATFORM(WPE) here

zhani replied:
Fixed

carlosgcampos commented on Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
```diff
@@ -76,6 +76,7 @@
 #include "WebWorkerClient.h"
 #include <WebCore/AppHighlight.h>
 #include <WebCore/AXObjectCache.h>
+#include <WebCore/BarcodeDetectorInterface.h>
```
This looks unrelated to GBM or LIBDRM, I guess it's a unified build issue that can be fixed in separate commit without review.

zhani replied:
I don't have commit rights to push without review. I added this change as a separate commit to this patch so it gets merged with this

carlosgcampos reviewed (changes requested):
Thanks, I have a few minor comments

zhani committed:
[WPE] Fix compilation issue with USE_GBM=OFF and USE_LIBDRM=OFF
https://bugs.webkit.org/show_bug.cgi?id=285229

Reviewed by Carlos Garcia Campos.

When USE_GBM=OFF and USE_LIBDRM=OFF, WPE platform compilation fails.
This combination also implies ENABLE_GPU_PROCESS=OFF. The patch
adds necessary ifdefs and includes to resolve the issue.

* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:

Canonical link: https://commits.webkit.org/288832@main
